### PR TITLE
chore: refactor out io-ts

### PR DESCRIPTION
**Pull request by frederickfogerty**
This is part of the Gatsby fp-ts refactor.

### CODE REVIEW

**Comment by frederickfogerty** [comment]
Current dependencies on/for this PR:
* main
  * **PR #198** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/198" target="_blank">Graphite</a> 
    * **PR #201** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/201" target="_blank">Graphite</a> 
      * **PR #202** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/202" target="_blank">Graphite</a> 
        * **PR #203** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/203" target="_blank">Graphite</a> 
          * **PR #204** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/204" target="_blank">Graphite</a> 
            * **PR #205** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/205" target="_blank">Graphite</a> 
              * **PR #206** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/206" target="_blank">Graphite</a> 
                * **PR #207** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/207" target="_blank">Graphite</a> 
                  * **PR #208** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/208" target="_blank">Graphite</a> 
                    * **PR #209** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/209" target="_blank">Graphite</a> 
                      * **PR #210** <a href="https://app.graphite.dev/github/pr/imgix/gatsby/210" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/imgix/gatsby/210?utm_source=stack-comment).

File: src/publicTypes.ts [hunk 1]
```diff
@@ -10,80 +10,94 @@ export enum ImgixSourceType {
...
+  .try(
+    Joi.string(),
+    Joi.number(),
+    Joi.boolean(),
+    Joi.array().items(Joi.string()),
+    Joi.array().items(Joi.number()),
+    Joi.array().items(Joi.boolean()),
+  )
+  .optional()
+  .allow(null);
```
Comment (sherwinski) [inline src/publicTypes.ts:35]: For folks unfamiliar with joi could we more details on what this and what it's supposed to do?

File: src/publicTypes.ts [hunk 2]
```diff
@@ -10,80 +10,94 @@ export enum ImgixSourceType {
...
-export type IImgixParams = t.TypeOf<typeof ImgixParamsIOTS>;
+): Record<TKey, typeof ImgixParamValueJOI> =>
+  mapObjIndexed(() => ImgixParamValueJOI, obj);
+
+const ImgixParamsJOI = Joi.object().keys({
+  ...mapToImgixParamJOIValue(imgixUrlParameters.aliases),
+  ...mapToImgixParamJOIValue(imgixUrlParameters.parameters),
+});
+
+export type IImgixParams = Partial<Record<IImgixParamsKey, ImgixParamValue>>;
```
Comment (sherwinski) [inline src/publicTypes.ts:56]: Likewise could we get some more details on what this is for

**Review by sherwinski — COMMENTED** [review]
✅

**Pull request merged**